sql/types/json_test.go

```diff
@@ -212,6 +212,29 @@ func TestValuer(t *testing.T) {
 	require.Equal(t, `{"a": "one"}`, res)
 }
 
+func TestLazyJsonDocument(t *testing.T) {
+	testCases := []struct {
+		s    string
+		json interface{}
+	}{
+		{"\"1\"", "1"},
+		{"{\"a\": [1.0, null]}", map[string]any{"a": []any{1.0, nil}}},
```

Comment:
(super minor) backtick quoted strings in Go are handy for avoiding having to escape double quotes. 

```suggestion
		{`"1"`, "1"},
		{`{"a": [1.0, null]}`, map[string]any{"a": []any{1.0, nil}}},
```

Reply:
Done.